compiler/resolution/functionResolution.cpp

```diff
@@ -7601,6 +7639,17 @@ static void instantiate_default_constructor(FnSymbol* fn) {
         }
       }
     }
+
+    if (AggregateType* ct = toAggregateType(fn->_this->type)) {
+      if (ct->instantiationStyle == DEFINES_INITIALIZER &&
```

Comment:
Actually, we insert calls to the default constructor into our user defined constructors back in change_method_to_constructor (or whatever its name is).

That last statement is probably a good point.  This code was originally from when I thought I could support generics in the old style and so might not be necessary now.  I'll try running without it.